Upgrading SelectiveTaint: Replacing dyninst9.3 with dyninst13.0 in Ubuntu 14.04 Environment

I'm trying to replace dynisnt9.3 with dyninst13.0 in selectivetaint (running in ubuntu14.04 environment) and I'm getting the following error in the build from source short step, so that means that dynisnt9.3 onwards is not supported for installation in the ubuntu14.04 (because of its i686 architecture)?

`
CMake Error at cmake/DyninstPlatform.cmake:16 (message):
  Unsupported architecture: 'i686'
Call Stack (most recent call first):
  CMakeLists.txt:35 (include)
`

I apologize for not responding to your professional answer in a timely manner. The patch you provided did resolve the issue.
However, when I run `cmake /path/to/dyninst/source -DCMAKE_INSTALL_PREFIX=/path/to/installation` , it reports `gcc>=6.0 (requested)`, which doesn't seem to recognize the gcc version correctly (so I added `-DCMAKE_C _COMPILER=/usr/bin/gcc-9 -DCMAKE_CXX_COMPILER=/usr/bin/g++-9` solved the problem). Then it started looking for third-party dependencies one by one, but cmake didn't download them automatically when it couldn't find them, so I had to manually source-compile and install the appropriate version of each dependency (`boost-1.78` `tbb-2019` compiled without any problems). When I was installing `elfutils-0.186`, it passed the `. /configure`, but at the make stage, the following error occurs, and I'm not sure if this old kernel version of Ubuntu doesn't support such a high version of elfutils. I'm a bit clueless about this error. I'm really sorry for disturbing your precious time again.


the output of `make` is as follows:

```
make --no-print-directory all-recursive
Making all in config
make[2]: Nothing to be done for `all'.
Making all in lib
make[2]: Nothing to be done for `all'.
Making all in libelf
make[2]: Nothing to be done for `all'.
Making all in libcpu
make[2]: Nothing to be done for `all'.
Making all in backends
make[2]: Nothing to be done for `all'.
Making all in libebl
make[2]: Nothing to be done for `all'.
Making all in libdwelf
  CC       dwelf_elf_gnu_build_id.o
  CC       dwelf_elf_begin.o
  AR       libdwelf.a
  CC       dwelf_elf_gnu_build_id.os
  CC       dwelf_elf_begin.os
  AR       libdwelf_pic.a
Making all in libdwfl
  CC       dwfl_begin.o
  CC       dwfl_end.o
  CC       dwfl_error.o
  CC       dwfl_version.o
  CC       dwfl_module.o
  CC       dwfl_report_elf.o
  CC       relocate.o
  CC       dwfl_module_build_id.o
  CC       dwfl_module_report_build_id.o
  CC       derelocate.o
  CC       offline.o
  CC       segment.o
  CC       dwfl_module_info.o
  CC       dwfl_getmodules.o
  CC       dwfl_getdwarf.o
  CC       dwfl_module_getdwarf.o
  CC       dwfl_module_getelf.o
  CC       dwfl_validate_address.o
  CC       argp-std.o
  CC       find-debuginfo.o
  CC       dwfl_build_id_find_elf.o
  CC       dwfl_build_id_find_debuginfo.o
  CC       linux-kernel-modules.o
  CC       linux-proc-maps.o
  CC       dwfl_addrmodule.o
  CC       dwfl_addrdwarf.o
  CC       cu.o
  CC       dwfl_module_nextcu.o
  CC       dwfl_nextcu.o
  CC       dwfl_cumodule.o
  CC       dwfl_module_addrdie.o
  CC       dwfl_addrdie.o
  CC       lines.o
  CC       dwfl_lineinfo.o
  CC       dwfl_line_comp_dir.o
  CC       dwfl_linemodule.o
  CC       dwfl_linecu.o
  CC       dwfl_dwarf_line.o
  CC       dwfl_getsrclines.o
  CC       dwfl_onesrcline.o
  CC       dwfl_module_getsrc.o
  CC       dwfl_getsrc.o
  CC       dwfl_module_getsrc_file.o
  CC       libdwfl_crc32.o
  CC       libdwfl_crc32_file.o
  CC       elf-from-memory.o
  CC       dwfl_module_dwarf_cfi.o
  CC       dwfl_module_eh_cfi.o
  CC       dwfl_module_getsym.o
  CC       dwfl_module_addrname.o
  CC       dwfl_module_addrsym.o
  CC       dwfl_module_return_value_location.o
  CC       dwfl_module_register_names.o
  CC       dwfl_segment_report_module.o
  CC       link_map.o
  CC       core-file.o
  CC       open.o
  CC       image-header.o
  CC       dwfl_frame.o
  CC       frame_unwind.o
  CC       dwfl_frame_pc.o
  CC       linux-pid-attach.o
  CC       linux-core-attach.o
  CC       dwfl_frame_regs.o
  CC       gzip.o
  CC       bzip2.o
  CC       debuginfod-client.o
  AR       libdwfl.a
  CC       dwfl_begin.os
  CC       dwfl_end.os
  CC       dwfl_error.os
  CC       dwfl_version.os
  CC       dwfl_module.os
  CC       dwfl_report_elf.os
  CC       relocate.os
  CC       dwfl_module_build_id.os
  CC       dwfl_module_report_build_id.os
  CC       derelocate.os
  CC       offline.os
  CC       segment.os
  CC       dwfl_module_info.os
  CC       dwfl_getmodules.os
  CC       dwfl_getdwarf.os
  CC       dwfl_module_getdwarf.os
  CC       dwfl_module_getelf.os
  CC       dwfl_validate_address.os
  CC       argp-std.os
  CC       find-debuginfo.os
  CC       dwfl_build_id_find_elf.os
  CC       dwfl_build_id_find_debuginfo.os
  CC       linux-kernel-modules.os
  CC       linux-proc-maps.os
  CC       dwfl_addrmodule.os
  CC       dwfl_addrdwarf.os
  CC       cu.os
  CC       dwfl_module_nextcu.os
  CC       dwfl_nextcu.os
  CC       dwfl_cumodule.os
  CC       dwfl_module_addrdie.os
  CC       dwfl_addrdie.os
  CC       lines.os
  CC       dwfl_lineinfo.os
  CC       dwfl_line_comp_dir.os
  CC       dwfl_linemodule.os
  CC       dwfl_linecu.os
  CC       dwfl_dwarf_line.os
  CC       dwfl_getsrclines.os
  CC       dwfl_onesrcline.os
  CC       dwfl_module_getsrc.os
  CC       dwfl_getsrc.os
  CC       dwfl_module_getsrc_file.os
  CC       libdwfl_crc32.os
  CC       libdwfl_crc32_file.os
  CC       elf-from-memory.os
  CC       dwfl_module_dwarf_cfi.os
  CC       dwfl_module_eh_cfi.os
  CC       dwfl_module_getsym.os
  CC       dwfl_module_addrname.os
  CC       dwfl_module_addrsym.os
  CC       dwfl_module_return_value_location.os
  CC       dwfl_module_register_names.os
  CC       dwfl_segment_report_module.os
  CC       link_map.os
  CC       core-file.os
  CC       open.os
  CC       image-header.os
  CC       dwfl_frame.os
  CC       frame_unwind.os
  CC       dwfl_frame_pc.os
  CC       linux-pid-attach.os
  CC       linux-core-attach.os
  CC       dwfl_frame_regs.os
  CC       gzip.os
  CC       bzip2.os
  CC       debuginfod-client.os
  AR       libdwfl_pic.a
Making all in libdw
make  all-am
  AR       libdw.a
  CCLD     libdw.so
Making all in libasm
  CCLD     libasm.so
Making all in debuginfod
  CXX      debuginfod.o
In file included from /usr/include/error.h:52,
                 from ../lib/system.h:46,
                 from debuginfod.cxx:40:
/usr/include/i386-linux-gnu/bits/error.h: In function ‘void error(int, int, const char*, ...)’:
/usr/include/i386-linux-gnu/bits/error.h:42:19: error: invalid use of ‘__builtin_va_arg_pack ()’
   42 |     __error_alias (__status, __errnum, __format, __va_arg_pack ());
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [debuginfod.o] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```



the output of `. /configure` is as follows:

```
configure: No --program-prefix given, using "eu-"
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... (cached) yes
checking build system type... i686-pc-linux-gnu
checking host system type... i686-pc-linux-gnu
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for gcc option to accept ISO C99... none needed
checking for ranlib... ranlib
checking for bison... no
checking for byacc... no
checking for flex... no
checking for lex... no
checking for ar... ar
checking the archiver (ar) interface... ar
checking for readelf... readelf
checking for nm... nm
checking whether gcc supports __attribute__((visibility()))... yes
checking whether gcc supports __attribute__((gcc_struct))... yes
checking whether gcc supports -fPIC... yes
checking whether gcc supports -fPIE... yes
checking whether gcc supports -Wl,-z,defs... yes
checking whether the compiler generates build-ids... yes
checking whether gcc supports -Wl,-z,relro... yes
checking for __thread support... yes
checking whether gcc provides stdatomic.h... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... 64
checking whether fts.h is bad when included (with LFS)... yes
checking whether to add -D_FORTIFY_SOURCE=2 to CFLAGS... yes
checking for library containing gzdirect... -lz
checking for library containing BZ2_bzdopen... -lbz2
checking for library containing lzma_auto_decoder... no
checking for library containing ZSTD_decompress... no
checking whether memrchr is declared... yes
checking whether rawmemchr is declared... yes
checking whether powerof2 is declared... yes
checking whether mempcpy is declared... yes
checking whether reallocarray is declared... no
checking for process_vm_readv... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking error.h usability... yes
checking error.h presence... yes
checking for error.h... yes
checking err.h usability... yes
checking err.h presence... yes
checking for err.h... yes
checking whether strerror_r is declared... yes
checking for strerror_r... yes
checking whether strerror_r returns char *... yes
checking for __cxa_demangle in -lstdc++... yes
checking whether symbol versioning is supported... yes
checking whether gcc accepts -Wstack-usage... yes
checking whether gcc has a sane -Wlogical-op... yes
checking whether gcc accepts -Wduplicated-cond... yes
checking whether gcc accepts -Wnull-dereference... yes
checking whether gcc accepts -Wimplicit-fallthrough... yes
checking whether the compiler accepts -Wimplicit-fallthrough=5... yes
checking whether the compiler accepts -Wtrampolines... yes
checking whether the compiler accepts -Wno-packed-not-aligned... yes
checking for library containing argp_parse... none required
checking for library containing fts_close... none required
checking for library containing _obstack_free... none required
checking for a sed that does not truncate output... /bin/sed
checking whether NLS is requested... yes
checking for msgfmt... /usr/bin/msgfmt
checking for gmsgfmt... /usr/bin/msgfmt
checking for xgettext... /usr/bin/xgettext
checking for msgmerge... /usr/bin/msgmerge
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for shared library run path origin... done
checking 32-bit host C ABI... yes
checking for ELF binary format... yes
checking for the common suffixes of directories in the library search path... lib,lib,lib
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyPreferredLanguages... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
checking size of long... 4
checking for struct user_regs_struct... yes
checking gcc option for 32-bit word size... -m32
checking for 64-bit host... no
checking whether gcc -m64 makes executables we can run... no
checking for bunzip2... yes
checking for zstd... no
configure: checking libdebuginfod dependencies, --disable-libdebuginfod or --enable-libdebuginfo=dummy to skip
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libcurl >= 7.29.0... yes
checking for pthread_setname_np in -lpthread... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
configure: checking debuginfod C++11 support, --disable-debuginfod to skip
checking whether g++ supports C++11 features by default... yes
configure: checking debuginfod dependencies, --disable-debuginfod to skip
checking for pkg-config... (cached) /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libmicrohttpd >= 0.9.33... yes
checking for sqlite3 >= 3.7.17... yes
checking for libarchive >= 3.1.2... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating config/Makefile
config.status: creating Makefile
config.status: creating elfutils.spec
config.status: creating debuginfod/Makefile
config.status: creating debuginfod/debuginfod.h
config.status: creating doc/Makefile
config.status: creating lib/Makefile
config.status: creating libelf/Makefile
config.status: creating libebl/Makefile
config.status: creating libdwelf/Makefile
config.status: creating libdw/Makefile
config.status: creating libdwfl/Makefile
config.status: creating libcpu/Makefile
config.status: creating libasm/Makefile
config.status: creating backends/Makefile
config.status: creating src/Makefile
config.status: creating po/Makefile.in
config.status: creating tests/Makefile
config.status: creating config/libelf.pc
config.status: creating config/libdw.pc
config.status: creating config/libdebuginfod.pc
config.status: creating version.h
config.status: creating config/profile.sh
config.status: creating config/profile.csh
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing po-directories commands
config.status: creating po/POTFILES
config.status: creating po/Makefile
configure:
=====================================================================
        elfutils: 0.186 (eu_version: 186)
=====================================================================

    Prefix                             : /usr/local
    Program prefix ("eu-" recommended) : eu-
    Source code location               : .
    Maintainer mode                    : 
    build arch                         : i686-pc-linux-gnu

  RECOMMENDED FEATURES (should all be yes)
    gzip support                       : yes
    bzip2 support                      : yes
    lzma/xz support                    : no
    zstd support                       : no
    libstdc++ demangle support         : yes
    File textrel check                 : yes
    Symbol versioning                  : yes

  NOT RECOMMENDED FEATURES (should all be no)
    Experimental thread safety         : no
    install elf.h                      : no

  OTHER FEATURES
    Deterministic archives by default  : false
    Native language support            : yes
    Extra Valgrind annotations         : no
    libdebuginfod client support       : yes
    Debuginfod server support          : yes
    Default DEBUGINFOD_URLS            : 

  EXTRA TEST FEATURES (used with make check)
    have bunzip2 installed (required)  : yes
    have zstd installed                : no
    debug branch prediction            : no
    gprof support                      : no
    gcov support                       : no
    run all tests under valgrind       : no
    gcc undefined behaviour sanitizer  : no
    use rpath in tests                 : no
    test biarch                        : no

```

I apologize for not responding to your professional answer in a timely manner. The patch you provided did resolve the issue.
However, when I run `cmake /path/to/dyninst/source -DCMAKE_INSTALL_PREFIX=/path/to/installation` , it reports `gcc>=6.0 (requested)`, which doesn't seem to recognize the gcc version correctly (so I added `-DCMAKE_C _COMPILER=/usr/bin/gcc-9 -DCMAKE_CXX_COMPILER=/usr/bin/g++-9` solved the problem). Then it started looking for third-party dependencies one by one, but cmake didn't download them automatically when it couldn't find them, so I had to manually source-compile and install the appropriate version of each dependency (`boost-1.78` `tbb-2019` compiled without any problems). When I was installing `elfutils-0.186`, it passed the `. /configure`, but at the make stage, the following error occurs, and I'm not sure if this old kernel version of Ubuntu doesn't support such a high version of elfutils. I'm a bit clueless about this error. I'm really sorry for disturbing your precious time again.


the output of `make` is as follows:

```
make --no-print-directory all-recursive
Making all in config
make[2]: Nothing to be done for `all'.
Making all in lib
make[2]: Nothing to be done for `all'.
Making all in libelf
make[2]: Nothing to be done for `all'.
Making all in libcpu
make[2]: Nothing to be done for `all'.
Making all in backends
make[2]: Nothing to be done for `all'.
Making all in libebl
make[2]: Nothing to be done for `all'.
Making all in libdwelf
  CC       dwelf_elf_gnu_build_id.o
  CC       dwelf_elf_begin.o
  AR       libdwelf.a
  CC       dwelf_elf_gnu_build_id.os
  CC       dwelf_elf_begin.os
  AR       libdwelf_pic.a
Making all in libdwfl
  CC       dwfl_begin.o
  CC       dwfl_end.o
  CC       dwfl_error.o
  CC       dwfl_version.o
  CC       dwfl_module.o
  CC       dwfl_report_elf.o
  CC       relocate.o
  CC       dwfl_module_build_id.o
  CC       dwfl_module_report_build_id.o
  CC       derelocate.o
  CC       offline.o
  CC       segment.o
  CC       dwfl_module_info.o
  CC       dwfl_getmodules.o
  CC       dwfl_getdwarf.o
  CC       dwfl_module_getdwarf.o
  CC       dwfl_module_getelf.o
  CC       dwfl_validate_address.o
  CC       argp-std.o
  CC       find-debuginfo.o
  CC       dwfl_build_id_find_elf.o
  CC       dwfl_build_id_find_debuginfo.o
  CC       linux-kernel-modules.o
  CC       linux-proc-maps.o
  CC       dwfl_addrmodule.o
  CC       dwfl_addrdwarf.o
  CC       cu.o
  CC       dwfl_module_nextcu.o
  CC       dwfl_nextcu.o
  CC       dwfl_cumodule.o
  CC       dwfl_module_addrdie.o
  CC       dwfl_addrdie.o
  CC       lines.o
  CC       dwfl_lineinfo.o
  CC       dwfl_line_comp_dir.o
  CC       dwfl_linemodule.o
  CC       dwfl_linecu.o
  CC       dwfl_dwarf_line.o
  CC       dwfl_getsrclines.o
  CC       dwfl_onesrcline.o
  CC       dwfl_module_getsrc.o
  CC       dwfl_getsrc.o
  CC       dwfl_module_getsrc_file.o
  CC       libdwfl_crc32.o
  CC       libdwfl_crc32_file.o
  CC       elf-from-memory.o
  CC       dwfl_module_dwarf_cfi.o
  CC       dwfl_module_eh_cfi.o
  CC       dwfl_module_getsym.o
  CC       dwfl_module_addrname.o
  CC       dwfl_module_addrsym.o
  CC       dwfl_module_return_value_location.o
  CC       dwfl_module_register_names.o
  CC       dwfl_segment_report_module.o
  CC       link_map.o
  CC       core-file.o
  CC       open.o
  CC       image-header.o
  CC       dwfl_frame.o
  CC       frame_unwind.o
  CC       dwfl_frame_pc.o
  CC       linux-pid-attach.o
  CC       linux-core-attach.o
  CC       dwfl_frame_regs.o
  CC       gzip.o
  CC       bzip2.o
  CC       debuginfod-client.o
  AR       libdwfl.a
  CC       dwfl_begin.os
  CC       dwfl_end.os
  CC       dwfl_error.os
  CC       dwfl_version.os
  CC       dwfl_module.os
  CC       dwfl_report_elf.os
  CC       relocate.os
  CC       dwfl_module_build_id.os
  CC       dwfl_module_report_build_id.os
  CC       derelocate.os
  CC       offline.os
  CC       segment.os
  CC       dwfl_module_info.os
  CC       dwfl_getmodules.os
  CC       dwfl_getdwarf.os
  CC       dwfl_module_getdwarf.os
  CC       dwfl_module_getelf.os
  CC       dwfl_validate_address.os
  CC       argp-std.os
  CC       find-debuginfo.os
  CC       dwfl_build_id_find_elf.os
  CC       dwfl_build_id_find_debuginfo.os
  CC       linux-kernel-modules.os
  CC       linux-proc-maps.os
  CC       dwfl_addrmodule.os
  CC       dwfl_addrdwarf.os
  CC       cu.os
  CC       dwfl_module_nextcu.os
  CC       dwfl_nextcu.os
  CC       dwfl_cumodule.os
  CC       dwfl_module_addrdie.os
  CC       dwfl_addrdie.os
  CC       lines.os
  CC       dwfl_lineinfo.os
  CC       dwfl_line_comp_dir.os
  CC       dwfl_linemodule.os
  CC       dwfl_linecu.os
  CC       dwfl_dwarf_line.os
  CC       dwfl_getsrclines.os
  CC       dwfl_onesrcline.os
  CC       dwfl_module_getsrc.os
  CC       dwfl_getsrc.os
  CC       dwfl_module_getsrc_file.os
  CC       libdwfl_crc32.os
  CC       libdwfl_crc32_file.os
  CC       elf-from-memory.os
  CC       dwfl_module_dwarf_cfi.os
  CC       dwfl_module_eh_cfi.os
  CC       dwfl_module_getsym.os
  CC       dwfl_module_addrname.os
  CC       dwfl_module_addrsym.os
  CC       dwfl_module_return_value_location.os
  CC       dwfl_module_register_names.os
  CC       dwfl_segment_report_module.os
  CC       link_map.os
  CC       core-file.os
  CC       open.os
  CC       image-header.os
  CC       dwfl_frame.os
  CC       frame_unwind.os
  CC       dwfl_frame_pc.os
  CC       linux-pid-attach.os
  CC       linux-core-attach.os
  CC       dwfl_frame_regs.os
  CC       gzip.os
  CC       bzip2.os
  CC       debuginfod-client.os
  AR       libdwfl_pic.a
Making all in libdw
make  all-am
  AR       libdw.a
  CCLD     libdw.so
Making all in libasm
  CCLD     libasm.so
Making all in debuginfod
  CXX      debuginfod.o
In file included from /usr/include/error.h:52,
                 from ../lib/system.h:46,
                 from debuginfod.cxx:40:
/usr/include/i386-linux-gnu/bits/error.h: In function ‘void error(int, int, const char*, ...)’:
/usr/include/i386-linux-gnu/bits/error.h:42:19: error: invalid use of ‘__builtin_va_arg_pack ()’
   42 |     __error_alias (__status, __errnum, __format, __va_arg_pack ());
      |     ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [debuginfod.o] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
```



the output of `. /configure` is as follows:

```
configure: No --program-prefix given, using "eu-"
checking for a BSD-compatible install... /usr/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /bin/mkdir -p
checking for gawk... no
checking for mawk... mawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... no
checking whether make supports nested variables... (cached) yes
checking build system type... i686-pc-linux-gnu
checking host system type... i686-pc-linux-gnu
checking whether make supports the include directive... yes (GNU style)
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... gcc3
checking for gcc option to accept ISO C99... none needed
checking for ranlib... ranlib
checking for bison... no
checking for byacc... no
checking for flex... no
checking for lex... no
checking for ar... ar
checking the archiver (ar) interface... ar
checking for readelf... readelf
checking for nm... nm
checking whether gcc supports __attribute__((visibility()))... yes
checking whether gcc supports __attribute__((gcc_struct))... yes
checking whether gcc supports -fPIC... yes
checking whether gcc supports -fPIE... yes
checking whether gcc supports -Wl,-z,defs... yes
checking whether the compiler generates build-ids... yes
checking whether gcc supports -Wl,-z,relro... yes
checking for __thread support... yes
checking whether gcc provides stdatomic.h... yes
checking for special C compiler options needed for large files... no
checking for _FILE_OFFSET_BITS value needed for large files... 64
checking whether fts.h is bad when included (with LFS)... yes
checking whether to add -D_FORTIFY_SOURCE=2 to CFLAGS... yes
checking for library containing gzdirect... -lz
checking for library containing BZ2_bzdopen... -lbz2
checking for library containing lzma_auto_decoder... no
checking for library containing ZSTD_decompress... no
checking whether memrchr is declared... yes
checking whether rawmemchr is declared... yes
checking whether powerof2 is declared... yes
checking whether mempcpy is declared... yes
checking whether reallocarray is declared... no
checking for process_vm_readv... yes
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /bin/grep
checking for egrep... /bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking error.h usability... yes
checking error.h presence... yes
checking for error.h... yes
checking err.h usability... yes
checking err.h presence... yes
checking for err.h... yes
checking whether strerror_r is declared... yes
checking for strerror_r... yes
checking whether strerror_r returns char *... yes
checking for __cxa_demangle in -lstdc++... yes
checking whether symbol versioning is supported... yes
checking whether gcc accepts -Wstack-usage... yes
checking whether gcc has a sane -Wlogical-op... yes
checking whether gcc accepts -Wduplicated-cond... yes
checking whether gcc accepts -Wnull-dereference... yes
checking whether gcc accepts -Wimplicit-fallthrough... yes
checking whether the compiler accepts -Wimplicit-fallthrough=5... yes
checking whether the compiler accepts -Wtrampolines... yes
checking whether the compiler accepts -Wno-packed-not-aligned... yes
checking for library containing argp_parse... none required
checking for library containing fts_close... none required
checking for library containing _obstack_free... none required
checking for a sed that does not truncate output... /bin/sed
checking whether NLS is requested... yes
checking for msgfmt... /usr/bin/msgfmt
checking for gmsgfmt... /usr/bin/msgfmt
checking for xgettext... /usr/bin/xgettext
checking for msgmerge... /usr/bin/msgmerge
checking for ld used by gcc... /usr/bin/ld
checking if the linker (/usr/bin/ld) is GNU ld... yes
checking for shared library run path origin... done
checking 32-bit host C ABI... yes
checking for ELF binary format... yes
checking for the common suffixes of directories in the library search path... lib,lib,lib
checking for CFPreferencesCopyAppValue... no
checking for CFLocaleCopyPreferredLanguages... no
checking for GNU gettext in libc... yes
checking whether to use NLS... yes
checking where the gettext function comes from... libc
checking size of long... 4
checking for struct user_regs_struct... yes
checking gcc option for 32-bit word size... -m32
checking for 64-bit host... no
checking whether gcc -m64 makes executables we can run... no
checking for bunzip2... yes
checking for zstd... no
configure: checking libdebuginfod dependencies, --disable-libdebuginfod or --enable-libdebuginfo=dummy to skip
checking for pkg-config... /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libcurl >= 7.29.0... yes
checking for pthread_setname_np in -lpthread... yes
checking for g++... g++
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... gcc3
configure: checking debuginfod C++11 support, --disable-debuginfod to skip
checking whether g++ supports C++11 features by default... yes
configure: checking debuginfod dependencies, --disable-debuginfod to skip
checking for pkg-config... (cached) /usr/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for libmicrohttpd >= 0.9.33... yes
checking for sqlite3 >= 3.7.17... yes
checking for libarchive >= 3.1.2... yes
checking that generated files are newer than configure... done
configure: creating ./config.status
config.status: creating config/Makefile
config.status: creating Makefile
config.status: creating elfutils.spec
config.status: creating debuginfod/Makefile
config.status: creating debuginfod/debuginfod.h
config.status: creating doc/Makefile
config.status: creating lib/Makefile
config.status: creating libelf/Makefile
config.status: creating libebl/Makefile
config.status: creating libdwelf/Makefile
config.status: creating libdw/Makefile
config.status: creating libdwfl/Makefile
config.status: creating libcpu/Makefile
config.status: creating libasm/Makefile
config.status: creating backends/Makefile
config.status: creating src/Makefile
config.status: creating po/Makefile.in
config.status: creating tests/Makefile
config.status: creating config/libelf.pc
config.status: creating config/libdw.pc
config.status: creating config/libdebuginfod.pc
config.status: creating version.h
config.status: creating config/profile.sh
config.status: creating config/profile.csh
config.status: creating config.h
config.status: config.h is unchanged
config.status: executing depfiles commands
config.status: executing po-directories commands
config.status: creating po/POTFILES
config.status: creating po/Makefile
configure:
=====================================================================
        elfutils: 0.186 (eu_version: 186)
=====================================================================

    Prefix                             : /usr/local
    Program prefix ("eu-" recommended) : eu-
    Source code location               : .
    Maintainer mode                    : 
    build arch                         : i686-pc-linux-gnu

  RECOMMENDED FEATURES (should all be yes)
    gzip support                       : yes
    bzip2 support                      : yes
    lzma/xz support                    : no
    zstd support                       : no
    libstdc++ demangle support         : yes
    File textrel check                 : yes
    Symbol versioning                  : yes

  NOT RECOMMENDED FEATURES (should all be no)
    Experimental thread safety         : no
    install elf.h                      : no

  OTHER FEATURES
    Deterministic archives by default  : false
    Native language support            : yes
    Extra Valgrind annotations         : no
    libdebuginfod client support       : yes
    Debuginfod server support          : yes
    Default DEBUGINFOD_URLS            : 

  EXTRA TEST FEATURES (used with make check)
    have bunzip2 installed (required)  : yes
    have zstd installed                : no
    debug branch prediction            : no
    gprof support                      : no
    gcov support                       : no
    run all tests under valgrind       : no
    gcc undefined behaviour sanitizer  : no
    use rpath in tests                 : no
    test biarch                        : no

```

To resolve the previous error (source install elfutils), apply the command `./configure --prefix=/usr --disable-debuginfod --enable-libdebuginfod=dummy` and that's it. However, after I continued to follow this link [8.48. Libelf from Elfutils-0.186 (linuxfromscratch.org)](https://www.linuxfromscratch.org/lfs/view/11.1/chapter08/libelf.html) and successfully installed libelf, running `cmake -DCMAKE_C_COMPILER=/usr/bin/gcc-9 -DCMAKE_CXX_ COMPILER=/usr/bin/g++-9 /home/dyninst/Downloads/dyninst-13.0.0 -DCMAKE_INSTALL_PREFIX=/home/dyninst/Downloads/dyninst_install` again, which produces the following Error. How can I manually set `LibELF_LIBRARIES` `LibELF_INCLUDE_DIRS` on the command line or somewhere else to ensure that cmake recognizes it.
```

-- C++ language standard:  11
-- C language standard:  11
-- Using C warning flags: -Wall;-Wextra;-Wpedantic;-Walloca;-Wcast-align;-Wcast-qual;-Wdouble-promotion;-Wduplicated-branches;-Wduplicated-cond;-Wfloat-equal;-Wformat-overflow=2;-Wformat-signedness;-Wformat=2;-Wframe-larger-than=20480;-Wjump-misses-init;-Wlogical-op;-Wmissing-braces;-Wmultichar;-Wpointer-arith;-Wrestrict;-Wshadow;-Wsuggest-attribute=format;-Wsuggest-attribute=malloc;-Wuninitialized;-Wvla;-Wwrite-strings
-- Using CXX warning flags: -Wall;-Wextra;-Wpedantic;-Walloca;-Wcast-align;-Wcast-qual;-Wctor-dtor-privacy;-Wdeprecated-copy-dtor;-Wdouble-promotion;-Wduplicated-branches;-Wduplicated-cond;-Wextra-semi;-Wfloat-equal;-Wformat-overflow=2;-Wformat-signedness;-Wformat=2;-Wframe-larger-than=20480;-Wlogical-op;-Wmissing-braces;-Wmultichar;-Wnoexcept;-Wnon-virtual-dtor;-Woverloaded-virtual;-Wpointer-arith;-Wrestrict;-Wshadow;-Wstrict-null-sentinel;-Wsuggest-attribute=format;-Wsuggest-attribute=malloc;-Wuninitialized;-Wvla;-Wwrite-strings
-- Extra CXX DEBUG warning flags: -Wframe-larger-than=24576
-- Found Boost 1.78.0
-- Boost include directories: /usr/include
-- Boost libraries: Boost::atomic;Boost::chrono;Boost::date_time;Boost::filesystem;Boost::thread;Boost::timer
-- Found TBB 2020.2
-- TBB include directories: /usr/local/include
-- Checking for module 'libelf>=0.186'
--   Found libelf, version 0.186
CMake Error at /usr/local/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find LibELF (missing: LibELF_LIBRARIES LibELF_INCLUDE_DIRS)
  (found suitable version "0.186", minimum required is "0.186")
Call Stack (most recent call first):
  /usr/local/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindLibELF.cmake:102 (find_package_handle_standard_args)
  cmake/Modules/FindElfutils.cmake:67 (find_package)
  cmake/tpls/DyninstElfUtils.cmake:39 (find_package)
  CMakeLists.txt:45 (include)


```

To resolve the previous error (source install elfutils), apply the command `./configure --prefix=/usr --disable-debuginfod --enable-libdebuginfod=dummy` and that's it. However, after I continued to follow this link [[8.48. Libelf from Elfutils-0.186 (linuxfromscratch.org)](https://www.linuxfromscratch.org/lfs/view/11.1/chapter08/libelf.html)](https://www.linuxfromscratch.org/lfs/view/11.1/chapter08/libelf.html) and successfully installed libelf, running `cmake -DCMAKE_C_COMPILER=/usr/bin/gcc-9 -DCMAKE_CXX_ COMPILER=/usr/bin/g++-9 /home/dyninst/Downloads/dyninst-13.0.0 -DCMAKE_INSTALL_PREFIX=/home/dyninst/Downloads/dyninst_install` again, which produces the following Error. How can I manually set `LibELF_LIBRARIES` `LibELF_INCLUDE_DIRS` on the command line or somewhere else to ensure that cmake recognizes it.

```
-- C++ language standard:  11
-- C language standard:  11
-- Using C warning flags: -Wall;-Wextra;-Wpedantic;-Walloca;-Wcast-align;-Wcast-qual;-Wdouble-promotion;-Wduplicated-branches;-Wduplicated-cond;-Wfloat-equal;-Wformat-overflow=2;-Wformat-signedness;-Wformat=2;-Wframe-larger-than=20480;-Wjump-misses-init;-Wlogical-op;-Wmissing-braces;-Wmultichar;-Wpointer-arith;-Wrestrict;-Wshadow;-Wsuggest-attribute=format;-Wsuggest-attribute=malloc;-Wuninitialized;-Wvla;-Wwrite-strings
-- Using CXX warning flags: -Wall;-Wextra;-Wpedantic;-Walloca;-Wcast-align;-Wcast-qual;-Wctor-dtor-privacy;-Wdeprecated-copy-dtor;-Wdouble-promotion;-Wduplicated-branches;-Wduplicated-cond;-Wextra-semi;-Wfloat-equal;-Wformat-overflow=2;-Wformat-signedness;-Wformat=2;-Wframe-larger-than=20480;-Wlogical-op;-Wmissing-braces;-Wmultichar;-Wnoexcept;-Wnon-virtual-dtor;-Woverloaded-virtual;-Wpointer-arith;-Wrestrict;-Wshadow;-Wstrict-null-sentinel;-Wsuggest-attribute=format;-Wsuggest-attribute=malloc;-Wuninitialized;-Wvla;-Wwrite-strings
-- Extra CXX DEBUG warning flags: -Wframe-larger-than=24576
-- Found Boost 1.78.0
-- Boost include directories: /usr/include
-- Boost libraries: Boost::atomic;Boost::chrono;Boost::date_time;Boost::filesystem;Boost::thread;Boost::timer
-- Found TBB 2020.2
-- TBB include directories: /usr/local/include
-- Checking for module 'libelf>=0.186'
--   Found libelf, version 0.186
CMake Error at /usr/local/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:230 (message):
  Could NOT find LibELF (missing: LibELF_LIBRARIES LibELF_INCLUDE_DIRS)
  (found suitable version "0.186", minimum required is "0.186")
Call Stack (most recent call first):
  /usr/local/share/cmake-3.27/Modules/FindPackageHandleStandardArgs.cmake:600 (_FPHSA_FAILURE_MESSAGE)
  cmake/Modules/FindLibELF.cmake:102 (find_package_handle_standard_args)
  cmake/Modules/FindElfutils.cmake:67 (find_package)
  cmake/tpls/DyninstElfUtils.cmake:39 (find_package)
  CMakeLists.txt:45 (include)
```

I checked the directories `/usr/include` and `/usr/lib`.

Under `/usr/include` is the folder `elfutils` with the following directory structure:

elfutils/
├── debuginfod.h
├── elf-knowledge.h
├── known-dwarf.h
├── libasm.h
├── libdwelf.h
├── libdwfl.h
├── libdw.h
└── version.h

Also in the `/usr/lib` directory I found `libelf.a` and `libelf-0.186.so`.

Here are the commands I ran when installing `elfutils`:

```
./configure --prefix=/usr --program-prefix="eu-" --disable-debuginfod --enable-libdebuginfod=dummy
make
make check
make install
```

The output after running `make` is as follows (I've omitted parts because it's too long):

```
make --no-print-directory all-recursive
Making all in config
make[2]: Nothing to be done for `all'.
Making all in lib
  CC       xasprintf.o
  CC       xstrdup.o
  CC       xstrndup.o
  CC       xmalloc.o
  CC       next_prime.o
  CC       crc32.o
  CC       crc32_file.o
  CC       color.o
  CC       error.o
  CC       printversion.o
  AR       libeu.a
Making all in libelf
  CC       elf_version.o
  CC       elf_hash.o
  CC       elf_error.o
...
  CC       elf_compress.os
  CC       elf_compress_gnu.os
  AR       libelf_pic.a
  CCLD     libelf.so
Making all in libcpu
  GEN      i386_defs
  GEN      i386.mnemonics
  CC       i386_disasm.o
  GEN      x86_64_defs
  GEN      x86_64.mnemonics
  CC       x86_64_disasm.o
  CC       bpf_disasm.o
  CC       riscv_disasm.o
  AR       libcpu.a
  CC       i386_disasm.os
  CC       x86_64_disasm.os
  CC       bpf_disasm.os
  CC       riscv_disasm.os
  AR       libcpu_pic.a
rm x86_64_defs i386_defs
Making all in backends
  CC       i386_init.o
  CC       i386_symbol.o
  CC       i386_corenote.o
  CC       i386_cfi.o
...
  CC       csky_regs.os
  CC       csky_initreg.os
  CC       csky_corenote.os
  AR       libebl_backends_pic.a
Making all in libebl
  CC       eblopenbackend.o
  CC       eblclosebackend.o
  CC       eblreloctypename.o
  CC       eblsegmenttypename.o
...
  CC       eblresolvesym.os
  CC       eblcheckreloctargettype.os
  CC       ebl_data_marker_symbol.os
  AR       libebl_pic.a
Making all in libdwelf
  CC       dwelf_elf_gnu_debuglink.o
  CC       dwelf_dwarf_gnu_debugaltlink.o
  CC       dwelf_elf_gnu_build_id.o
  CC       dwelf_scn_gnu_compressed_size.o
  CC       dwelf_strtab.o
  CC       dwelf_elf_begin.o
  CC       dwelf_elf_e_machine_string.o
  AR       libdwelf.a
  CC       dwelf_elf_gnu_debuglink.os
  CC       dwelf_dwarf_gnu_debugaltlink.os
  CC       dwelf_elf_gnu_build_id.os
  CC       dwelf_scn_gnu_compressed_size.os
  CC       dwelf_strtab.os
  CC       dwelf_elf_begin.os
  CC       dwelf_elf_e_machine_string.os
  AR       libdwelf_pic.a
Making all in libdwfl
  CC       dwfl_begin.o
  CC       dwfl_end.o
  CC       dwfl_error.o
  CC       dwfl_version.o
...
  CC       gzip.os
  CC       bzip2.os
  CC       debuginfod-client.os
  AR       libdwfl_pic.a
Making all in libdw
make  all-am
  CC       dwarf_begin.o
  CC       dwarf_begin_elf.o
  CC       dwarf_end.o
  CC       dwarf_getelf.o
...
  CC       dwarf_cu_info.os
  CC       dwarf_next_lines.os
  AR       libdw_pic.a
  CCLD     libdw.so
Making all in libasm
  CC       asm_begin.o
  CC       asm_abort.o
  CC       asm_end.o
  CC       asm_error.o
...
  CC       disasm_str.os
  CC       symbolhash.os
  AR       libasm_pic.a
  CCLD     libasm.so
Making all in debuginfod
  CC       debuginfod-find.o
  CC       debuginfod-client.o
  CC       debuginfod-client.os
  AR       libdebuginfod_pic.a
  CCLD     libdebuginfod.so.1
ln -fs libdebuginfod.so.1 libdebuginfod.so
  CCLD     debuginfod-find
  AR       libdebuginfod.a
Making all in src
  CC       readelf.o
  CCLD     readelf
  CC       nm.o
  CCLD     nm
...
  CCLD     elfcompress
  CC       elfclassify.o
  CCLD     elfclassify
  GEN      make-debug-archive
Making all in po
make[2]: Nothing to be done for `all'.
Making all in doc
make[2]: Nothing to be done for `all'.
Making all in tests
make[2]: Nothing to be done for `all'.
```

The complete output after running `make check` is as follows:

```
Making check in config
make[1]: Nothing to be done for `check'.
Making check in lib
make[1]: Nothing to be done for `check'.
Making check in libelf
make[1]: Nothing to be done for `check'.
Making check in libcpu
make[1]: Nothing to be done for `check'.
Making check in backends
make[1]: Nothing to be done for `check'.
Making check in libebl
make[1]: Nothing to be done for `check'.
Making check in libdwelf
make[1]: Nothing to be done for `check'.
Making check in libdwfl
make[1]: Nothing to be done for `check'.
Making check in libdw
make  check-am
make[2]: Nothing to be done for `check-am'.
Making check in libasm
make[1]: Nothing to be done for `check'.
Making check in debuginfod
make[1]: Nothing to be done for `check'.
Making check in src
make[1]: Nothing to be done for `check'.
Making check in po
make[1]: Nothing to be done for `check'.
Making check in doc
make[1]: Nothing to be done for `check'.
Making check in tests
make  arextract arsymtest newfile saridx scnnames sectiondump showptable update1 update2 update3 update4 test-nlist show-die-info get-files next-files get-lines next-lines get-pubnames get-aranges allfcts line2addr addrscopes funcscopes show-abbrev hash newscn ecp dwflmodtest find-prologues funcretval allregs rdwrmmap dwfl-bug-addr-overflow arls dwfl-bug-fd-leak dwfl-addr-sect dwfl-bug-report early-offscn dwfl-bug-getmodules dwarf-getmacros dwarf-ranges addrcfi dwarfcfi test-flag-nobits dwarf-getstring rerequest_tag alldts typeiter typeiter2 low_high_pc test-elf_cntl_gelf_getshdr dwflsyms dwfllines dwfl-report-elf-align dwfl-report-segment-contiguous varlocs backtrace backtrace-child backtrace-data backtrace-dwarf debuglink debugaltlink buildid deleted deleted-lib.so aggregate_size peel_type vdsosyms getsrc_die strptr newdata elfstrtab dwfl-proc-attach elfshphehdr elfstrmerge dwelfgnucompressed elfgetchdr elfgetzdata elfputzdata zstrptr emptyfile vendorelf fillfile dwarf_default_lower_bound dwarf-die-addr-die get-units-invalid get-units-split attr-integrate-skel all-dwarf-ranges unit-info next_cfi elfcopy addsections xlate_notes elfrdwrnop dwelf_elf_e_machine_string getphdrnum leb128 read_unaligned msg_tst system-elf-libelf-test nvidia_extended_linemap_libdw asm-tst1 asm-tst2 asm-tst3 asm-tst4 asm-tst5 asm-tst6 asm-tst7 asm-tst8 asm-tst9  
  CC       arextract.o
  CCLD     arextract
  CC       arsymtest.o
  CCLD     arsymtest
  CC       newfile.o
  CCLD     newfile
  CC       saridx.o
  CCLD     saridx
  CC       scnnames.o
  CCLD     scnnames
  CC       sectiondump.o
  CCLD     sectiondump
  CC       showptable.o
  CCLD     showptable
  CC       update1.o
  CCLD     update1
  CC       update2.o
  CCLD     update2
  CC       update3.o
  CCLD     update3
  CC       update4.o
  CCLD     update4
  CC       test-nlist
  CC       show-die-info.o
  CCLD     show-die-info
  CC       get-files.o
  CCLD     get-files
  CC       next-files.o
  CCLD     next-files
  CC       get-lines.o
  CCLD     get-lines
  CC       next-lines.o
  CCLD     next-lines
  CC       get-pubnames.o
  CCLD     get-pubnames
  CC       get-aranges.o
  CCLD     get-aranges
  CC       allfcts.o
  CCLD     allfcts
  CC       line2addr.o
  CCLD     line2addr
  CC       addrscopes.o
  CCLD     addrscopes
  CC       funcscopes.o
  CCLD     funcscopes
  CC       show-abbrev.o
  CCLD     show-abbrev
  CC       hash.o
  CCLD     hash
  CC       newscn.o
  CCLD     newscn
  CC       ecp.o
  CCLD     ecp
  CC       dwflmodtest.o
  CCLD     dwflmodtest
  CC       find-prologues.o
  CCLD     find-prologues
  CC       funcretval.o
  CCLD     funcretval
  CC       allregs.o
  CCLD     allregs
  CC       rdwrmmap.o
  CCLD     rdwrmmap
  CC       dwfl-bug-addr-overflow.o
  CCLD     dwfl-bug-addr-overflow
  CC       arls.o
  CCLD     arls
  CC       dwfl-bug-fd-leak.o
  CCLD     dwfl-bug-fd-leak
  CC       dwfl-addr-sect.o
  CCLD     dwfl-addr-sect
  CC       dwfl-bug-report.o
  CCLD     dwfl-bug-report
  CC       early-offscn.o
  CCLD     early-offscn
  CC       dwfl-bug-getmodules.o
  CCLD     dwfl-bug-getmodules
  CC       dwarf-getmacros.o
  CCLD     dwarf-getmacros
  CC       dwarf-ranges.o
  CCLD     dwarf-ranges
  CC       addrcfi.o
  CCLD     addrcfi
  CC       dwarfcfi.o
  CCLD     dwarfcfi
  CC       test-flag-nobits.o
  CCLD     test-flag-nobits
  CC       dwarf-getstring.o
  CCLD     dwarf-getstring
  CC       rerequest_tag.o
  CCLD     rerequest_tag
  CC       alldts.o
  CCLD     alldts
  CC       typeiter.o
  CCLD     typeiter
  CC       typeiter2.o
  CCLD     typeiter2
  CC       low_high_pc.o
  CCLD     low_high_pc
  CC       test-elf_cntl_gelf_getshdr.o
  CCLD     test-elf_cntl_gelf_getshdr
  CC       dwflsyms.o
  CCLD     dwflsyms
  CC       dwfllines.o
  CCLD     dwfllines
  CC       dwfl-report-elf-align.o
  CCLD     dwfl-report-elf-align
  CC       dwfl-report-segment-contiguous.o
  CCLD     dwfl-report-segment-contiguous
  CC       varlocs.o
  CCLD     varlocs
  CC       backtrace.o
  CCLD     backtrace
  CC       backtrace_child-backtrace-child.o
  CCLD     backtrace-child
  CC       backtrace-data.o
  CCLD     backtrace-data
  CC       backtrace_dwarf-backtrace-dwarf.o
  CCLD     backtrace-dwarf
  CC       debuglink.o
  CCLD     debuglink
  CC       debugaltlink.o
  CCLD     debugaltlink
  CC       buildid.o
  CCLD     buildid
  CC       deleted.o
  CC       deleted_lib_so-deleted-lib.o
  CCLD     deleted-lib.so
  CCLD     deleted
make[2]: `deleted-lib.so' is up to date.
  CC       aggregate_size.o
  CCLD     aggregate_size
  CC       peel_type.o
  CCLD     peel_type
  CC       vdsosyms.o
  CCLD     vdsosyms
  CC       getsrc_die.o
  CCLD     getsrc_die
  CC       strptr.o
  CCLD     strptr
  CC       newdata.o
  CCLD     newdata
  CC       elfstrtab.o
  CCLD     elfstrtab
  CC       dwfl-proc-attach.o
  CCLD     dwfl-proc-attach
  CC       elfshphehdr.o
  CCLD     elfshphehdr
  CC       elfstrmerge.o
  CCLD     elfstrmerge
  CC       dwelfgnucompressed.o
  CCLD     dwelfgnucompressed
  CC       elfgetchdr.o
  CCLD     elfgetchdr
  CC       elfgetzdata.o
  CCLD     elfgetzdata
  CC       elfputzdata.o
  CCLD     elfputzdata
  CC       zstrptr.o
  CCLD     zstrptr
  CC       emptyfile.o
  CCLD     emptyfile
  CC       vendorelf.o
  CCLD     vendorelf
  CC       fillfile.o
  CCLD     fillfile
  CC       dwarf_default_lower_bound.o
  CCLD     dwarf_default_lower_bound
  CC       dwarf-die-addr-die.o
  CCLD     dwarf-die-addr-die
  CC       get-units-invalid.o
  CCLD     get-units-invalid
  CC       get-units-split.o
  CCLD     get-units-split
  CC       attr-integrate-skel.o
  CCLD     attr-integrate-skel
  CC       all-dwarf-ranges.o
  CCLD     all-dwarf-ranges
  CC       unit-info.o
  CCLD     unit-info
  CC       next_cfi.o
  CCLD     next_cfi
  CC       elfcopy.o
  CCLD     elfcopy
  CC       addsections.o
  CCLD     addsections
  CC       xlate_notes.o
  CCLD     xlate_notes
  CC       elfrdwrnop.o
  CCLD     elfrdwrnop
  CC       dwelf_elf_e_machine_string.o
  CCLD     dwelf_elf_e_machine_string
  CC       getphdrnum.o
  CCLD     getphdrnum
  CC       leb128.o
  CCLD     leb128
  CC       read_unaligned.o
  CCLD     read_unaligned
  CC       msg_tst.o
  CCLD     msg_tst
  CC       system_elf_libelf_test-system-elf-libelf-test.o
  CCLD     system-elf-libelf-test
  CC       nvidia_extended_linemap_libdw.o
  CCLD     nvidia_extended_linemap_libdw
  CC       asm-tst1.o
  CCLD     asm-tst1
  CC       asm-tst2.o
  CCLD     asm-tst2
  CC       asm-tst3.o
  CCLD     asm-tst3
  CC       asm-tst4.o
  CCLD     asm-tst4
  CC       asm-tst5.o
  CCLD     asm-tst5
  CC       asm-tst6.o
  CCLD     asm-tst6
  CC       asm-tst7.o
  CCLD     asm-tst7
  CC       asm-tst8.o
  CCLD     asm-tst8
  CC       asm-tst9.o
  CCLD     asm-tst9
make  check-TESTS
PASS: run-arextract.sh
PASS: run-arsymtest.sh
PASS: run-ar.sh
PASS: newfile
PASS: test-nlist
PASS: update1
PASS: update2
PASS: update3
PASS: update4
PASS: run-show-die-info.sh
PASS: run-get-files.sh
PASS: run-get-lines.sh
PASS: run-next-files.sh
PASS: run-next-lines.sh
PASS: run-get-pubnames.sh
PASS: run-get-aranges.sh
PASS: run-allfcts.sh
PASS: run-show-abbrev.sh
PASS: run-line2addr.sh
PASS: hash
SKIP: run-large-elf-file.sh
PASS: newscn
PASS: run-strip-test.sh
PASS: run-strip-test2.sh
PASS: run-strip-test3.sh
PASS: run-strip-test4.sh
PASS: run-strip-test5.sh
PASS: run-strip-test6.sh
PASS: run-strip-test7.sh
PASS: run-strip-test8.sh
PASS: run-strip-test9.sh
PASS: run-strip-test10.sh
PASS: run-strip-test11.sh
PASS: run-strip-test12.sh
PASS: run-strip-nothing.sh
PASS: run-strip-g.sh
PASS: run-annobingroup.sh
PASS: run-strip-groups.sh
PASS: run-strip-reloc.sh
PASS: run-strip-strmerge.sh
PASS: run-strip-nobitsalign.sh
PASS: run-strip-remove-keep.sh
PASS: run-unstrip-test.sh
PASS: run-unstrip-test2.sh
PASS: run-unstrip-test3.sh
PASS: run-unstrip-test4.sh
PASS: run-unstrip-M.sh
PASS: run-elfstrmerge-test.sh
PASS: run-ecp-test.sh
PASS: run-ecp-test2.sh
PASS: run-alldts.sh
PASS: run-elflint-test.sh
PASS: run-elflint-self.sh
PASS: run-ranlib-test.sh
PASS: run-ranlib-test2.sh
PASS: run-ranlib-test3.sh
PASS: run-ranlib-test4.sh
PASS: run-addrscopes.sh
PASS: run-strings-test.sh
PASS: run-funcscopes.sh
PASS: run-find-prologues.sh
PASS: run-allregs.sh
PASS: run-addrcfi.sh
PASS: run-dwarfcfi.sh
PASS: run-nm-syms.sh
PASS: run-nm-self.sh
PASS: run-readelf-self.sh
PASS: run-readelf-info-plus.sh
PASS: run-readelf-compressed.sh
PASS: run-readelf-const-values.sh
PASS: run-varlocs-self.sh
PASS: run-exprlocs-self.sh
PASS: run-readelf-test1.sh
PASS: run-readelf-test2.sh
PASS: run-readelf-test3.sh
PASS: run-readelf-test4.sh
PASS: run-readelf-twofiles.sh
PASS: run-readelf-macro.sh
PASS: run-readelf-loc.sh
PASS: run-readelf-ranges.sh
PASS: run-readelf-aranges.sh
PASS: run-readelf-line.sh
PASS: run-readelf-z.sh
PASS: run-readelf-frames.sh
PASS: run-readelf-n.sh
PASS: run-retain.sh
PASS: run-native-test.sh
PASS: run-bug1-test.sh
PASS: run-debuglink.sh
PASS: run-debugaltlink.sh
PASS: run-buildid.sh
PASS: dwfl-bug-addr-overflow
PASS: run-addrname-test.sh
PASS: dwfl-bug-fd-leak
PASS: dwfl-bug-report
PASS: dwfl-report-segment-contiguous
PASS: run-dwfl-bug-offline-rel.sh
PASS: run-dwfl-addr-sect.sh
PASS: run-disasm-x86.sh
PASS: run-disasm-x86-64.sh
PASS: run-early-offscn.sh
PASS: run-dwarf-getmacros.sh
PASS: run-dwarf-ranges.sh
PASS: run-test-flag-nobits.sh
PASS: run-prelink-addr-test.sh
PASS: run-dwarf-getstring.sh
PASS: run-rerequest_tag.sh
PASS: run-typeiter.sh
PASS: run-readelf-d.sh
PASS: run-readelf-gdb_index.sh
PASS: run-unstrip-n.sh
PASS: run-low_high_pc.sh
PASS: run-macro-test.sh
PASS: run-elf_cntl_gelf_getshdr.sh
PASS: run-test-archive64.sh
PASS: run-readelf-vmcoreinfo.sh
PASS: run-readelf-mixed-corenote.sh
PASS: run-dwfllines.sh
PASS: run-readelf-variant.sh
PASS: run-readelf-fat-lto.sh
PASS: run-dwfl-report-elf-align.sh
PASS: run-addr2line-test.sh
PASS: run-addr2line-i-test.sh
PASS: run-addr2line-i-lex-test.sh
PASS: run-addr2line-i-demangle-test.sh
PASS: run-addr2line-alt-debugpath.sh
PASS: run-varlocs.sh
PASS: run-exprlocs.sh
PASS: run-varlocs-vars.sh
PASS: run-funcretval.sh
PASS: run-backtrace-native.sh
SKIP: run-backtrace-data.sh
PASS: run-backtrace-dwarf.sh
SKIP: run-backtrace-native-biarch.sh
PASS: run-backtrace-native-core.sh
SKIP: run-backtrace-native-core-biarch.sh
PASS: run-backtrace-core-x86_64.sh
PASS: run-backtrace-fp-core-x86_64.sh
PASS: run-backtrace-fp-core-aarch64.sh
PASS: run-backtrace-fp-core-ppc64le.sh
PASS: run-backtrace-core-x32.sh
PASS: run-backtrace-core-i386.sh
PASS: run-backtrace-fp-core-i386.sh
PASS: run-backtrace-core-ppc.sh
PASS: run-backtrace-core-s390x.sh
PASS: run-backtrace-core-s390.sh
PASS: run-backtrace-core-aarch64.sh
PASS: run-backtrace-core-sparc.sh
PASS: run-backtrace-demangle.sh
PASS: run-stack-d-test.sh
PASS: run-stack-i-test.sh
PASS: run-stack-demangled-test.sh
PASS: run-readelf-zx.sh
PASS: run-readelf-zp.sh
PASS: run-readelf-addr.sh
PASS: run-readelf-str.sh
PASS: run-readelf-multi-noline.sh
PASS: run-readelf-types.sh
PASS: run-readelf-dwz-multi.sh
PASS: run-allfcts-multi.sh
PASS: run-deleted.sh
PASS: run-linkmap-cut.sh
PASS: run-aggregate-size.sh
PASS: run-peel-type.sh
PASS: vdsosyms
PASS: run-readelf-A.sh
PASS: run-getsrc-die.sh
PASS: run-strptr.sh
PASS: newdata
PASS: elfstrtab
PASS: dwfl-proc-attach
PASS: elfshphehdr
FAIL: run-lfs-symbols.sh
PASS: run-dwelfgnucompressed.sh
PASS: run-elfgetchdr.sh
PASS: run-elfgetzdata.sh
PASS: run-elfputzdata.sh
PASS: run-zstrptr.sh
PASS: run-compress-test.sh
PASS: run-readelf-zdebug.sh
PASS: run-readelf-zdebug-rel.sh
PASS: emptyfile
PASS: vendorelf
PASS: fillfile
PASS: dwarf_default_lower_bound
PASS: run-dwarf-die-addr-die.sh
PASS: run-get-units-invalid.sh
PASS: run-get-units-split.sh
PASS: run-attr-integrate-skel.sh
PASS: run-all-dwarf-ranges.sh
PASS: run-unit-info.sh
PASS: run-reloc-bpf.sh
PASS: run-next-cfi.sh
PASS: run-next-cfi-self.sh
PASS: run-reverse-sections.sh
PASS: run-reverse-sections-self.sh
PASS: run-copyadd-sections.sh
PASS: run-copymany-sections.sh
PASS: run-typeiter-many.sh
PASS: run-strip-test-many.sh
PASS: run-strip-version.sh
PASS: run-xlate-note.sh
PASS: run-readelf-discr.sh
PASS: run-dwelf_elf_e_machine_string.sh
PASS: run-elfclassify.sh
PASS: run-elfclassify-self.sh
PASS: run-disasm-riscv64.sh
PASS: run-pt_gnu_prop-tests.sh
PASS: run-getphdrnum.sh
PASS: run-test-includes.sh
PASS: leb128
PASS: read_unaligned
PASS: msg_tst
PASS: system-elf-libelf-test
PASS: asm-tst1
PASS: asm-tst2
PASS: asm-tst3
PASS: asm-tst4
PASS: asm-tst5
PASS: asm-tst6
PASS: asm-tst7
PASS: asm-tst8
PASS: asm-tst9
PASS: run-disasm-bpf.sh
PASS: run-low_high_pc-dw-form-indirect.sh
PASS: run-nvidia-extended-linemap-libdw.sh
PASS: run-nvidia-extended-linemap-readelf.sh
PASS: run-readelf-dw-form-indirect.sh
PASS: run-strip-largealign.sh
============================================================================
Testsuite summary for elfutils 0.186
============================================================================
# TOTAL: 229
# PASS:  224
# SKIP:  4
# XFAIL: 0
# FAIL:  1
# XPASS: 0
# ERROR: 0
============================================================================
See tests/test-suite.log
Please report to https://sourceware.org/bugzilla
============================================================================
make[3]: *** [test-suite.log] Error 1
make[2]: *** [check-TESTS] Error 2
make[1]: *** [check-am] Error 2
make: *** [check-recursive] Error 1
```

Here's a `FAIL: run-lfs-symbols.sh` I didn't bother with.

Finally, the output after running `make install` as root is as follows:

```
Making install in config
make[2]: Nothing to be done for `install-exec-am'.
/usr/bin/install -c -m 644 profile.sh -D /usr/etc/profile.d/debuginfod.sh
/usr/bin/install -c -m 644 profile.csh -D /usr/etc/profile.d/debuginfod.csh
mkdir -p /usr/etc/debuginfod
if [ -n "" ]; then \
		echo "" > /usr/etc/debuginfod/elfutils.urls; \
	fi
 /bin/mkdir -p '/usr/lib/pkgconfig'
 /usr/bin/install -c -m 644 libelf.pc libdw.pc libdebuginfod.pc '/usr/lib/pkgconfig'
Making install in lib
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in libelf
 /bin/mkdir -p '/usr/lib'
 /usr/bin/install -c -m 644  libelf.a '/usr/lib'
 ( cd '/usr/lib' && ranlib libelf.a )
 /bin/mkdir -p '/usr/include'
 /usr/bin/install -c -m 644 libelf.h gelf.h nlist.h '/usr/include'
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 elf-knowledge.h '/usr/include/elfutils'
/bin/bash /home/dyninst/Downloads/elfutils-0.186/config/install-sh -d /usr/lib
/usr/bin/install -c libelf.so /usr/lib/libelf-0.186.so
ln -fs libelf-0.186.so /usr/lib/libelf.so.1
ln -fs libelf.so.1 /usr/lib/libelf.so
Making install in libcpu
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in backends
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in libebl
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
Making install in libdwelf
make[2]: Nothing to be done for `install-exec-am'.
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 libdwelf.h '/usr/include/elfutils'
Making install in libdwfl
make[2]: Nothing to be done for `install-exec-am'.
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 libdwfl.h '/usr/include/elfutils'
Making install in libdw
 /bin/mkdir -p '/usr/lib'
 /usr/bin/install -c -m 644  libdw.a '/usr/lib'
 ( cd '/usr/lib' && ranlib libdw.a )
 /bin/mkdir -p '/usr/include'
 /usr/bin/install -c -m 644 dwarf.h '/usr/include'
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 libdw.h known-dwarf.h '/usr/include/elfutils'
/bin/bash /home/dyninst/Downloads/elfutils-0.186/config/install-sh -d /usr/lib
/usr/bin/install -c libdw.so /usr/lib/libdw-0.186.so
ln -fs libdw-0.186.so /usr/lib/libdw.so.1
ln -fs libdw.so.1 /usr/lib/libdw.so
Making install in libasm
 /bin/mkdir -p '/usr/lib'
 /usr/bin/install -c -m 644  libasm.a '/usr/lib'
 ( cd '/usr/lib' && ranlib libasm.a )
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 libasm.h '/usr/include/elfutils'
/bin/bash /home/dyninst/Downloads/elfutils-0.186/config/install-sh -d /usr/lib
/usr/bin/install -c libasm.so /usr/lib/libasm-0.186.so
ln -fs libasm-0.186.so /usr/lib/libasm.so.1
ln -fs libasm.so.1 /usr/lib/libasm.so
Making install in debuginfod
 /bin/mkdir -p '/usr/bin'
  /usr/bin/install -c debuginfod-find '/usr/bin'
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 debuginfod.h '/usr/include/elfutils'
/bin/bash /home/dyninst/Downloads/elfutils-0.186/config/install-sh -d /usr/lib
/usr/bin/install -c libdebuginfod.so.1 \
		/usr/lib/libdebuginfod-0.186.so
ln -fs libdebuginfod-0.186.so /usr/lib/libdebuginfod.so.1
ln -fs libdebuginfod-0.186.so /usr/lib/libdebuginfod.so
Making install in src
 /bin/mkdir -p '/usr/bin'
  /usr/bin/install -c readelf '/usr/bin/./eu-readelf'
  /usr/bin/install -c nm '/usr/bin/./eu-nm'
  /usr/bin/install -c size '/usr/bin/./eu-size'
  /usr/bin/install -c strip '/usr/bin/./eu-strip'
  /usr/bin/install -c elflint '/usr/bin/./eu-elflint'
  /usr/bin/install -c findtextrel '/usr/bin/./eu-findtextrel'
  /usr/bin/install -c addr2line '/usr/bin/./eu-addr2line'
  /usr/bin/install -c elfcmp '/usr/bin/./eu-elfcmp'
  /usr/bin/install -c objdump '/usr/bin/./eu-objdump'
  /usr/bin/install -c ranlib '/usr/bin/./eu-ranlib'
  /usr/bin/install -c strings '/usr/bin/./eu-strings'
  /usr/bin/install -c ar '/usr/bin/./eu-ar'
  /usr/bin/install -c unstrip '/usr/bin/./eu-unstrip'
  /usr/bin/install -c stack '/usr/bin/./eu-stack'
  /usr/bin/install -c elfcompress '/usr/bin/./eu-elfcompress'
  /usr/bin/install -c elfclassify '/usr/bin/./eu-elfclassify'
 /bin/mkdir -p '/usr/bin'
 /usr/bin/install -c make-debug-archive '/usr/bin/./eu-make-debug-archive'
make[2]: Nothing to be done for `install-data-am'.
Making install in po
installing de.gmo as /usr/share/locale/de/LC_MESSAGES/elfutils.mo
installing es.gmo as /usr/share/locale/es/LC_MESSAGES/elfutils.mo
installing ja.gmo as /usr/share/locale/ja/LC_MESSAGES/elfutils.mo
installing pl.gmo as /usr/share/locale/pl/LC_MESSAGES/elfutils.mo
installing uk.gmo as /usr/share/locale/uk/LC_MESSAGES/elfutils.mo
installing en@quot.gmo as /usr/share/locale/en@quot/LC_MESSAGES/elfutils.mo
installing en@boldquot.gmo as /usr/share/locale/en@boldquot/LC_MESSAGES/elfutils.mo
if test "elfutils" = "gettext-tools"; then \
	  /bin/mkdir -p /usr/share/gettext/po; \
	  for file in Makefile.in.in remove-potcdate.sin quot.sed boldquot.sed en@quot.header en@boldquot.header insert-header.sin Rules-quot   Makevars.template; do \
	    /usr/bin/install -c -m 644 ./$file \
			    /usr/share/gettext/po/$file; \
	  done; \
	  for file in Makevars; do \
	    rm -f /usr/share/gettext/po/$file; \
	  done; \
	else \
	  : ; \
	fi
Making install in doc
make[2]: Nothing to be done for `install-exec-am'.
 /bin/mkdir -p '/usr/share/man/man1'
 /usr/bin/install -c -m 644 debuginfod-find.1 '/usr/share/man/man1'
 /bin/mkdir -p '/usr/share/man/man1'
 /usr/bin/install -c -m 644 'readelf.1' '/usr/share/man/man1/eu-readelf.1'
 /usr/bin/install -c -m 644 'elfclassify.1' '/usr/share/man/man1/eu-elfclassify.1'
 /bin/mkdir -p '/usr/share/man/man3'
 /usr/bin/install -c -m 644 elf_update.3 elf_getdata.3 elf_clone.3 elf_begin.3 debuginfod_add_http_header.3 debuginfod_begin.3 debuginfod_end.3 debuginfod_find_debuginfo.3 debuginfod_find_executable.3 debuginfod_find_source.3 debuginfod_get_user_data.3 debuginfod_get_url.3 debuginfod_set_progressfn.3 debuginfod_set_user_data.3 '/usr/share/man/man3'
 /bin/mkdir -p '/usr/share/man/man7'
 /usr/bin/install -c -m 644 debuginfod-client-config.7 '/usr/share/man/man7'
Making install in tests
make[2]: Nothing to be done for `install-exec-am'.
make[2]: Nothing to be done for `install-data-am'.
make[2]: Nothing to be done for `install-exec-am'.
 /bin/mkdir -p '/usr/include/elfutils'
 /usr/bin/install -c -m 644 version.h '/usr/include/elfutils'
```

To summarize, maybe I missed some key commands during the installation of `elfutils`? Or maybe the `FAIL: run-lfs-symbols.sh` generated during the `make check` had an impact on the smooth installation of `elfutils`?

Execute `cmake -DCMAKE_C_COMPILER=/usr/bin/gcc-9 -DCMAKE_CXX_COMPILER=/usr/bin/g++-9 /home/dyninst/Downloads/dyninst-13.0.0 -DCMAKE_INSTALL_PREFIX=/home/dyninst/Downloads/dyninst_install -DElfUtils_NO_SYSTEM_PATHS=ON -DElfUtils_ROOT_DIR=/usr > output_cmake.txt 2>&1`  The output is as follows:

```
-- C++ language standard:  11
-- C language standard:  11
-- Using C warning flags: -Wall;-Wextra;-Wpedantic;-Walloca;-Wcast-align;-Wcast-qual;-Wdouble-promotion;-Wduplicated-branches;-Wduplicated-cond;-Wfloat-equal;-Wformat-overflow=2;-Wformat-signedness;-Wformat=2;-Wframe-larger-than=20480;-Wjump-misses-init;-Wlogical-op;-Wmissing-braces;-Wmultichar;-Wpointer-arith;-Wrestrict;-Wshadow;-Wsuggest-attribute=format;-Wsuggest-attribute=malloc;-Wuninitialized;-Wvla;-Wwrite-strings
-- Using CXX warning flags: -Wall;-Wextra;-Wpedantic;-Walloca;-Wcast-align;-Wcast-qual;-Wctor-dtor-privacy;-Wdeprecated-copy-dtor;-Wdouble-promotion;-Wduplicated-branches;-Wduplicated-cond;-Wextra-semi;-Wfloat-equal;-Wformat-overflow=2;-Wformat-signedness;-Wformat=2;-Wframe-larger-than=20480;-Wlogical-op;-Wmissing-braces;-Wmultichar;-Wnoexcept;-Wnon-virtual-dtor;-Woverloaded-virtual;-Wpointer-arith;-Wrestrict;-Wshadow;-Wstrict-null-sentinel;-Wsuggest-attribute=format;-Wsuggest-attribute=malloc;-Wuninitialized;-Wvla;-Wwrite-strings
-- Extra CXX DEBUG warning flags: -Wframe-larger-than=24576
-- Found Boost 1.78.0
-- Boost include directories: /usr/include
-- Boost libraries: Boost::atomic;Boost::chrono;Boost::date_time;Boost::filesystem;Boost::thread;Boost::timer
-- Found TBB 2020.2
-- TBB include directories: /usr/local/include
CMake Error at cmake/Modules/FindLibELF.cmake:61 (pkg_get_variable):
  Unknown CMake command "pkg_get_variable".
Call Stack (most recent call first):
  cmake/Modules/FindElfutils.cmake:67 (find_package)
  cmake/tpls/DyninstElfUtils.cmake:39 (find_package)
  CMakeLists.txt:45 (include)


-- Configuring incomplete, errors occurred!
```

It still seems to be hard to fix, I'm thinking of switching to ubuntu 16.04 32-bit or downgrading dynisnt later on.

--- a/cmake/DyninstPlatform.cmake
+++ b/cmake/DyninstPlatform.cmake
@@ -11,17 +11,25 @@ endif()
 # Equivalent to CMAKE_HOST_SYSTEM_PROCESSOR and `uname -m` on Unixes
 cmake_host_system_information(RESULT _host_arch QUERY OS_PLATFORM)
 
-set(_known_arches "x86_64" "ppc64le" "aarch64" "i386" "amd64")
+set(_32bit_x86_arches "i386" "i686")
+set(_64bit_x86_arches "x86_64" "amd64")
+set(_known_arches "ppc64le" "aarch64" ${_32bit_x86_arches} ${_64bit_x86_arches})
+
 if(NOT ${_host_arch} IN_LIST _known_arches)
   message(FATAL_ERROR "Unsupported architecture: '${_host_arch}'")
+endif()
+
+set(_is_32bit_x86 FALSE)
+if(_host_arch IN_LIST _32bit_x86_arches)
+  set(_is_32bit_x86 TRUE)
 endif()
 
 # Equivalent to checking CMAKE_SIZEOF_VOID_P
 cmake_host_system_information(RESULT _is64bit QUERY IS_64BIT)
 
-# 32-bit is only supported on i386
-if(NOT _is64bit AND NOT ${_host_arch} STREQUAL "i386")
-  message(FATAL_ERROR "32-bit programming is only supported on i386")
+# 32-bit is only supported on 32-bit x86
+if(NOT _is64bit AND NOT ${_is_32bit_x86})
+  message(FATAL_ERROR "32-bit programming is only supported on ${_32bit_x86_arches}")
 endif()
 
 # These checks are redundant, but protect against string name changes
@@ -39,13 +47,13 @@ if(DYNINST_OS_Linux OR DYNINST_OS_FreeBSD)
   set(DYNINST_OS_UNIX TRUE)
 endif()
 
-if(${_host_arch} STREQUAL "x86_64" OR ${_host_arch} STREQUAL "amd64")
+if(${_host_arch} IN_LIST _64bit_x86_arches)
   set(DYNINST_ARCH_x86_64 TRUE)
 elseif(${_host_arch} STREQUAL "aarch64")
   set(DYNINST_ARCH_aarch64 TRUE)
 elseif(${_host_arch} STREQUAL "ppc64le")
   set(DYNINST_ARCH_ppc64le TRUE)
-elseif(${_host_arch} STREQUAL "i386")
+elseif(${_host_arch} IN_LIST _32bit_x86_arches)
   set(DYNINST_ARCH_i386 TRUE)
 endif()
 
@@ -55,7 +63,7 @@ if(${_host_os} STREQUAL "Linux")
   if(NOT _is64bit)
     set(DYNINST_PLATFORM i386-unknown-linux2.4)
   else()
-    if(${_host_arch} STREQUAL "x86_64")
+    if(${_host_arch} IN_LIST _64bit_x86_arches)
       set(DYNINST_PLATFORM x86_64-unknown-linux2.4)
     elseif(${_host_arch} STREQUAL "aarch64")
       set(DYNINST_PLATFORM aarch64-unknown-linux)

